lib/algos/list/delimiterMaster.js

```diff
@@ -131,6 +128,22 @@ class DelimiterMaster extends Delimiter {
         }
 
         this.prvKey = key;
+        if (key === this.prefix) {
+            // key is of the form foo/
+            this.delimiterEndingKeySeen = true;
```

Comment:
The variable name does not seem to match what it's supposed to mean, as it could refer to any key with an ending delimiter, where it's set only if we've seen a key that `is` the prefix to list. If this is needed, why not calling this variable `keyMatchingPrefixSeen` for example instead.